src/coreclr/jit/compiler.cpp

```diff
@@ -5352,12 +5359,19 @@ PhaseStatus Compiler::placeLoopAlignInstructions()
                 madeChanges       = true;
                 unmarkedLoopAlign = true;
             }
-            else if ((block->bbNatLoopNum != BasicBlock::NOT_IN_LOOP) && (block->bbNatLoopNum == loopTop->bbNatLoopNum))
+            else if (visitedLoopNum[loopTop->bbNatLoopNum])
```

Comment:
Don't you need to check that `block->bbNatLoopNum != BasicBlock::NOT_IN_LOOP`?

Reply:
Not really. `block` is the previous block before `loopTop` and we want to check here if we have seen a loop body related to `loopTop->bbNatLoopNum`.

Reply:
Oh, sorry, I meant, don't you need to check `loopTop->bbNatLoopNum != BasicBlock::NOT_IN_LOOP`?

It appears that sometimes it does equal NOT_IN_LOOP, leading to reading beyond the array bounds.